java/java.mx.project/test/unit/src/org/netbeans/modules/java/mx/project/SuiteCheck.java

```diff
@@ -0,0 +1,257 @@
+                    if (sourceGroup.getDisplayName().equals(gName)) {
+                        cnt++;
+                        break FOUND;
+                    }
+                }
+                // not found
+                continue;
+            }
+            assertSourcesNoError(p, errornous, errors, sourceGroup.getRootFolder(), begin);
+        }
+        assertCompilationErrors(errors, errornous);
+
+        assertEquals("Exactly as many source groups tested as requested", onlySourceGroups.length, cnt);
+    }
+
+    protected final File findSuite(String suite) throws URISyntaxException {
+        File location = getDataDir();
+        while (location != null) {
+            File graal = new File(location, "graal");
+            File suiteDir = new File(graal, suite);
+            if (suiteDir.isDirectory()) {
+                return suiteDir;
+            }
+            location = location.getParentFile();
+        }
+        fail("Cannot find truffle next to " + getDataDir());
+        return null;
+    }
+
+    private void assertSourcesNoError(Project project, FileObject[] errornous, StringBuilder errors, FileObject dir, long begin) throws IOException {
+        long now = System.currentTimeMillis();
+        log(Level.INFO, "assertSourcesNoError for %s", dir);
+        IndexingManager.getDefault().refreshIndexAndWait(dir.toURL(), null, false);
+        log(Level.INFO, "      refresh done       %s", dir);
+        Enumeration<? extends FileObject> en = dir.getChildren(true);
+        int nonJavaCount = 0;
+        int javaCount = 0;
+        while (en.hasMoreElements()) {
+            FileObject fo = en.nextElement();
+            if (fo.isFolder()) {
+                continue;
+            }
+            Project prj = FileOwnerQuery.getOwner(fo);
+            assertSame("FileOwnerQuery returns the right project", project, prj);
+            if (!fo.hasExt("java")) {
+                nonJavaCount++;
+                continue;
+            }
+            JavaSource source = JavaSource.forFileObject(fo);
+            if (source == null) {
+                fail("No source for " + fo);
+            }
+            javaCount++;
+            BinaryForSourceQuery.Result res = BinaryForSourceQuery.findBinaryRoots(dir.toURL());
+            assertEquals("There is one binary root: " + Arrays.toString(res.getRoots()), 1, res.getRoots().length);
+            OK: for (URL root : res.getRoots()) {
+                SourceForBinaryQuery.Result2 res2 = SourceForBinaryQuery.findSourceRoots2(root);
+                assertTrue("Has to prefer sources", res2.preferSources());
+                for (FileObject src : res2.getRoots()) {
+                    if (src.equals(dir)) {
+                        break OK;
+                    }
+                }
+                fail("Expecting to find " + dir + " among:\n" + Arrays.toString(res2.getRoots()));
+            }
+            source.runUserActionTask((CompilationController p) -> {
```

Comment:
Is there a better way to verify project sources are fine from a NetBeans Java Source support perspective?